### PR TITLE
Avoids makeEventSink in RealGameWorkflow

### PR DESCRIPTION
**Pull request by rjrjr**
Just too much to digest in a presentation.

### CODE REVIEW

File: app/src/main/java/com/example/gameworkflow/Point.kt [hunk 1]
```diff
@@ -2,12 +2,17 @@ package com.example.gameworkflow
 
 data class Point(val x: Int, val y: Int)
 
+fun Point.constrainTo(bounds: Point): Point = copy(
```
Comment (rjrjr) [inline app/src/main/java/com/example/gameworkflow/Point.kt]: Suppose I could make the first two fun's private.

File: app/src/main/java/com/example/gameworkflow/RealGameWorkflow.kt [hunk 2]
```diff
@@ -2,39 +2,49 @@ package com.example.gameworkflow
...
-        }
+        val sink: Sink<WorkflowAction<State, GameOutput>> = context.makeActionSink()
+        val doMove = { direction: Direction -> sink.send(doMove(props, direction)) }
 
         return GameRendering(
             boardSize = props.boardSize,
             playerPosition = state.playerPosition,
             goalPosition = props.goalPosition,
-            onMove = if (!isGameOver(state.playerPosition)) moveSink::send else null
+            onMove = if (state.gameOver) doMove else null
```
Comment (rjrjr) [inline app/src/main/java/com/example/gameworkflow/RealGameWorkflow.kt]: When I try to inline `doMove` the compiler goes berserk, rants about wanting a `Nothing?`.

**Comment by rjrjr** [comment]
Well shit, hold up, I broke the game somehow.

**Comment by rjrjr** [comment]
Fixed. Also reduced `sealed class GameOutput` to `object GameEnded`.

File: app/src/main/java/com/example/gameworkflow/Point.kt [hunk 3]
```diff
@@ -2,12 +2,17 @@ package com.example.gameworkflow
...
 )
 
-fun Point.constrainTo(bounds: Point): Point = copy(
-    x = x.coerceIn(0, bounds.x - 1),
-    y = y.coerceIn(0, bounds.y - 1)
-)
+fun Point.moved(
+    direction: Direction,
+    bounds: Point
+): Point = moved(direction).constrainTo(bounds)
```
Comment (zach-klippenstein) [inline app/src/main/java/com/example/gameworkflow/Point.kt:18]: Why this change?

File: app/src/main/java/com/example/gameworkflow/GameWorkflow.kt [hunk 4]
```diff
@@ -2,17 +2,15 @@ package com.example.gameworkflow
...
 data class GameProps(
     val boardSize: Point,
     val spawnPosition: Point,
     val goalPosition: Point
 )
 
-sealed class GameOutput {
-    object Win : GameOutput()
-}
+object GameEnded
```
Comment (zach-klippenstein) [inline app/src/main/java/com/example/gameworkflow/GameWorkflow.kt:13]: 👍

**Review by zach-klippenstein — APPROVED** [review]

File: app/src/main/java/com/example/gameworkflow/Point.kt (hunk 3 above)
Comment (rjrjr) [reply app/src/main/java/com/example/gameworkflow/Point.kt:18]: It's the only way we ever actually call it, and felt like one less bit of thinking required when reading the game workflow.

**Comment by rjrjr** [comment]
We might want to rebase and force push the branch s.t. this change is applied to the first commit as a fixup, but of course I won't do that behind your back. 

Safe flight.
